onyo/lib/inventory.py

```diff
@@ -725,3 +725,66 @@ def raise_empty_keys(self, asset: dict | UserDict) -> None:
             # Note, that DotNotationWrapper.keys() delivers strings (and has to).
             # Hence, `None` as a key would show up here as 'None'.
             raise ValueError("Keys are not allowed to be empty or None-values.")
+
+    def _parse_operations_record(self, record: list[str]):
+
+        if not record[0].strip() == "--- Inventory Operations ---":
+            raise RuntimeError("Invalid operations record.")
+        parsed_record = {k: [] for k in OPERATIONS_MAPPING.keys()}
+        collecting_key = None
+        for line in record[1:]:
+            line = line.strip()
+            if not line:
+                continue
+            match line:
+                case "New assets:":
+                    collecting_key = "new_assets"
+                case "New directories:":
+                    collecting_key = "new_directories"
+                case "Removed assets:":
+                    collecting_key = "remove_assets"
+                case "Removed directories:":
+                    collecting_key = "remove_directories"
+                case "Moved assets:":
+                    collecting_key = "move_assets"
+                case "Moved directories:":
+                    collecting_key = "move_directories"
+                case "Renamed directories:":
+                    collecting_key = "rename_directories"
+                case "Renamed assets:":
+                    collecting_key = "rename_assets"
+                case "Modified assets:":
+                    collecting_key = "modify_assets"
+                case _:
+                    if not collecting_key or line and not line.startswith("- "):
+                        raise RuntimeError("Invalid operations record.")
+                    cleaned_line = line[2:].split(" -> ")
+                    if len(cleaned_line) > 2:
+                        raise RuntimeError(f"Invalid operations record:{line}")
+                    if len(cleaned_line) == 1:
+                        parsed_record[collecting_key].append(Path(cleaned_line[0]))
+                    else:
+                        parsed_record[collecting_key].append(
+                            (Path(cleaned_line[0]), Path(cleaned_line[1]))
+                        )
+        return parsed_record
+
+    def get_history(self, path: Path | None = None, n: int | None = None):
+        # TODO: This isn't quite right yet. operations records are defined in Inventory.
+        #       But Inventory shouldn't talk to GitRepo directly. So, either pass the record
+        #       from Inventory to OnyoRepo and turn it into a commit-message part only,
+        #       or have sort of a proxy in OnyoRepo.
+        #       -> May be: get_history(Item) in Inventory and get_history(path) in OnyoRepo.
+
+        for commit in self.repo.git.follow(path, n):
```

Comment:
I know I'm on a bit of a regex trip, but wouldn't this be easier (and faster) to extract as the group after `--- Inventory Operations ---\n`?

Reply:
If it was one big string, then yes. Not convinced of that yet, though.
However, code may still be simpler by looking for that line and get the rest via list slicing.